cpp/CMakeLists.txt

```diff
@@ -357,6 +380,7 @@ target_link_libraries(cuopt_cli
   PUBLIC
   cuopt
   OpenMP::OpenMP_CXX
+  papilo-core
```

Comment:
I don't see papilo headers being exposed in cupot so this should be PRVIATE

Reply:
Done. Thanks!